Sources/idt/idt.cc

```diff
@@ -126,16 +126,41 @@ struct PPCallbacks : clang::PPCallbacks {
   FileIncludes &file_includes_;
 };
 
+// Track a set of clang::Decl declarations by unique ID.
+class DeclSet {
+  // Use the maximum allowed fixed-size for SmallPtrSet. If the set grows
+  // larger than this size, SmallPtrSet switches to "large set" behavior.
+  static constexpr size_t SMALL_SET_SIZE = 32;
+  llvm::SmallPtrSet<std::uintptr_t, SMALL_SET_SIZE> decls_;
```

Comment:
I don't think that the comment is needed - this is a pretty commonly used pattern in LLVM.

```suggestion
  llvm::SmallPtrSet<std::uintptr_t, 32> decls_;
```